pkg/controller/atlascluster/cluster.go

```diff
@@ -38,17 +44,66 @@ func ensureClusterState(wctx *workflow.Context, connection atlas.Connection, pro
 
 	switch c.StateName {
 	case "IDLE":
-		return c, workflow.OK()
+		spec, err := cluster.Spec.Cluster()
+		if err != nil {
+			return c, workflow.Terminate(workflow.Internal, err.Error())
+		}
+
+		if done, err := clusterMatchesSpec(wctx, c, cluster.Spec); err != nil {
+			return c, workflow.Terminate(workflow.Internal, err.Error())
+		} else if done {
+			return c, workflow.OK()
+		}
+
+		c, _, err = client.Clusters.Update(ctx, project.Status.ID, cluster.Spec.Name, spec)
+		if err != nil {
+			return c, workflow.Terminate(workflow.ClusterNotCreatedInAtlas, err.Error())
+		}
+
+		return c, workflow.InProgress(workflow.ClusterNotUpToDate, "cluster update started")
 
 	case "CREATING":
-		return c, workflow.InProgress(workflow.ClusterCreating, "Cluster is provisioning")
+		return c, workflow.InProgress(workflow.ClusterCreating, "cluster is provisioning")
 
 	case "UPDATING", "REPAIRING":
-		return c, workflow.InProgress(workflow.ClusterUpdating, "Cluster is updating")
+		return c, workflow.InProgress(workflow.ClusterUpdating, "cluster is updating")
 
 	// TODO: add "DELETING", "DELETED", handle 404 on delete
 
 	default:
 		return c, workflow.Terminate(workflow.Internal, fmt.Sprintf("unknown cluster state %q", c.StateName))
 	}
 }
+
+// clusterMatchesSpec will merge everything from the Spec into existing Cluster and use that to detect change
```

Comment:
[nit] a bit more information is welcome (e.g. why we are doing this and cannot compare objects directly - as the cluster from Atlas may have default/deprecated fields)